### ISSUE TITLE
error: this statement may fall through [-Werror=implicit-fallthrough=]

### ISSUE THREAD

**Issue by tmdfree** [michaelfm1211/mdview#1]
Hello, I failed to build mdview. 

Because of the build option  "-Werror" The warning turn to error. 
After remove the opton, it work fine. But I do not know if it is ok.

error message
```
cc -Wall -Wextra -Werror -std=c99 -pedantic -O2 -fvisibility=hidden -c -o lib/links.o lib/links.c
lib/links.c: In function ‘handle_link_special_char’:
lib/links.c:13:8: error: this statement may fall through [-Werror=implicit-fallthrough=]
   13 |     if (!ctx->pending_link) {
      |        ^
lib/links.c:18:3: note: here
   18 |   case '[':
      |   ^~~~
cc1: all warnings being treated as errors
make: *** [Makefile:25: lib/links.o] Error 1

```

gcc version

```
gcc (Debian 12.2.0-14) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by michaelfm1211** [comment]
Thanks for pointing this out!

Weird, neither clang or gcc-13 on macOS outputs this error. GCC's right, there should've been a `break` in that case. This should be fixed in 82f1937. If there's still an issue, please let me know.

**Comment by tmdfree** [comment]
Work fine now.
